cmd/thanos/query-frontend.go

```diff
@@ -111,51 +91,40 @@ func runQueryFrontend(
 	logger log.Logger,
 	reg *prometheus.Registry,
 	tracer opentracing.Tracer,
-	conf *queryFrontendConfig,
+	cfg *config,
 	comp component.Component,
 ) error {
 
-	if len(conf.downstreamURL) == 0 {
-		return errors.New("downstream URL should be configured")
+	err := cfg.Cache.Validate()
+	if err != nil {
+		return errors.Wrap(err, "error validating cache config")
+	}
+	err = cfg.QueryRange.Validate(logger)
+	if err != nil {
+		return errors.Wrap(err, "error validating query range config")
 	}
 
 	fe, err := frontend.New(frontend.Config{
-		DownstreamURL:        conf.downstreamURL,
-		CompressResponses:    conf.compressResponses,
-		LogQueriesLongerThan: conf.LogQueriesLongerThan,
+		DownstreamURL:        cfg.Frontend.DownstreamURL,
+		CompressResponses:    cfg.Frontend.CompressResponses,
+		LogQueriesLongerThan: cfg.Frontend.LogQueriesLongerThan,
 	}, logger, reg)
 	if err != nil {
 		return errors.Wrap(err, "setup query frontend")
 	}
 	defer fe.Close()
 
-	limits := queryfrontend.NewLimits(
-		conf.queryRangeConfig.maxQueryParallelism,
-		time.Duration(conf.queryRangeConfig.maxQueryLength),
-		conf.queryRangeConfig.cacheMaxFreshness,
-	)
-
-	respCacheContentYaml, err := conf.queryRangeConfig.respCacheConfig.Content()
+	limits, err := validation.NewOverrides(cfg.Limits, nil)
```

Comment:
It is not good to use Cortex's limits since it is too complex and you will find it adds more fields https://github.com/thanos-io/thanos/pull/3114/files#diff-d06787af93a387061cc1d2fe8067558eR87-R122 in the config file. 

Would you mind using our own? There are only 3 interface so not much code for us to maintain.

Reply:
Yep I thought the same thing but we can still reuse the `validation.NewOverrides` no?

Reply:
I think we can. If we need to extend the limits interface in the future, we can easily switch to our own.